engine/collection/compliance/core_test.go

```diff
@@ -286,7 +286,10 @@ func (cs *CoreSuite) TestOnBlockProposal_FailsHotStuffValidation() {
 		*cs.validator = *hotstuff.NewValidator(cs.T())
 		sentinelError := model.NewInvalidProposalErrorf(hotstuffProposal, "")
 		cs.validator.On("ValidateProposal", hotstuffProposal).Return(sentinelError)
-		cs.proposalViolationNotifier.On("OnInvalidBlockDetected", sentinelError).Return().Once()
+		cs.proposalViolationNotifier.On("OnInvalidBlockDetected", flow.Slashable[model.InvalidProposalError]{
+			OriginID: originID,
+			Message:  sentinelError.(model.InvalidProposalError),
```

Comment:
We need to cast this because `NewInvalidProposalErrorf` returns an `error` rather than `InvalidProposalError` type. I'm wondering if there's a reason why we do that? I feel like returning the concrete type and letting the caller interpret it as the appropriate interface type is usually more flexible.

Reply:
Yeah, that's true, but this pattern was used everywhere else so I wasn't sure why. I can change it for at least this one.